rdkPlugins/Networking/source/IPAllocator.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "IPAllocator.h"
+
+#include <Logging.h>
+
+#include <sys/stat.h>
+#include <dirent.h>
+#include <algorithm>
+#include <fcntl.h>
+#include <unistd.h>
+
+IPAllocator::IPAllocator(const std::shared_ptr<DobbyRdkPluginUtils> &utils)
+    : mUtils(utils)
+{
+    AI_LOG_FN_ENTRY();
+
+    // Update internal state based on the disk store
+    if (!updateFromStore())
+    {
+        AI_LOG_ERROR("Failed to initialise IP backing store");
+    }
+
+    // Start from xxx.xxx.xxx.2 to leave xxx.xxx.xxx.1 open for bridge device
+    in_addr_t addrBegin = INADDR_BRIDGE + 1;
+    in_addr_t addrEnd = addrBegin + TOTAL_ADDRESS_POOL_SIZE;
+
+    // Populate the pool of available addresses
```

Comment:
Should something be here? This newline suggest to me that you wanted to add some piece of code here.